printing/proof_diffs.ml

```diff
@@ -547,89 +536,79 @@ module GoalMap = Evar.Map
-  (match op with
-  | Some op ->
-    let (ogoals,_,_,_,osigma) = Proof.proof op in
-    Printf.printf "\nOld Goals: ";
-    List.iter (fun og -> Printf.printf "%d -> %s  " (Evar.repr og) (goal_to_evar og osigma)) ogoals
-  | None -> ());
-  Printf.printf "\nGoal map: ";
-  GoalMap.iter (fun og ng -> Printf.printf "%d -> %d  " (Evar.repr og) (Evar.repr ng)) ng_to_og;
-  Printf.printf "\n"
-[@@@ocaml.warning "+32"]
-
-(* Create a map from new goals to old goals for proof diff.  The map only
- has entries for new goals that are not the same as the corresponding old
- goal; there are no entries for unchanged goals.
-
- It proceeds as follows:
- 1. Find the goal ids that were removed from the old proof and that were
- added in the new proof.  If the same goal id is present in both proofs
- then conclude the goal is unchanged (assumption).
-
- 2. The code assumes that proof changes only take the form of replacing
- one or more goal symbols (CEvars) with new terms.  Therefore:
- - if there are no removals, the proofs are the same.
- - if there are removals but no additions, then there are no new goals
-   that aren't the same as their associated old goals.  For the both of
-   these cases, the map is empty because there are no new goals that differ
-   from their old goals
- - if there is only one removal, then any added goals should be mapped to
-   the removed goal.
- - if there are more than 2 removals and more than one addition, call
-   match_goals to get a map between old and new evar names, then use this
-   to create the map from new goal ids to old goal ids for the differing goals.
-*)
+(**
+   Create a map from new goals to old goals for proof diff.  The map
+    only has entries for new goals that are not the same as the
+    corresponding old goal; there are no entries for unchanged goals.
+
+    It proceeds as follows:
+
+    1. Find the goal ids that were removed from the old proof and that
+    were added in the new proof.  If the same goal id is present in
+    both proofs then conclude the goal is unchanged (assumption).
+
+    2. The code assumes that proof changes only take the form of
+    replacing one or more goal symbols (CEvars) with new terms.
+    Therefore:
+
+    - if there are no removals, the proofs are the same.
+
+    - if there are removals but no additions, then there are no new
+    goals that aren't the same as their associated old goals. For the
+    both of these cases, the map is empty because there are no new
+    goals that differ from their old goals
+
+    - if there is only one removal, then any added goals should be
+    mapped to the removed goal.
+
+    - if there are more than 2 removals and more than one addition,
+    call match_goals to get a map between old and new evar names, then
+    use this to create the map from new goal ids to old goal ids for
+    the differing goals. *)
+
 let make_goal_map_i op np =
   let ng_to_og = ref GoalMap.empty in
-  match op with
-  | None -> !ng_to_og
```

Comment:
The None case is necessary.  There is no old proof for the first state in the proof.  Where did that logic go?